accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,210 @@
+
+The new folder structure for NuGet plugins on the Windows platform is as follows: The `tools` folder contains .NET tool plugins, while the `.store` folder houses other necessary files.
+
+![plugin-tools-folder-windows](./../../meta/resources/PluginsAsDotNetTools/plugin-tools-folder-windows.png)
+
+![plugin-tools-store-windows](./../../meta/resources/PluginsAsDotNetTools/plugin-tools-store-windows.png)
+
+The new folder structure for NuGet plugins on the Linux platforms is as follows:
+
+```log
+{user}:~/.nuget/plugins$ dir
+netcore  tools
+
+{user}:~/.nuget/plugins$ cd tools/
+{user}:~/.nuget/plugins/tools$ dir
+botsay  dotnetsay
+
+{user}:~/.nuget/plugins/tools$ ls -la
+total 164
+drwxr-xr-x 3 {user}  4096 Feb 10 08:21 .
+drwxr-xr-x 4 {user}  4096 Feb 10 08:20 ..
+drwxr-xr-x 5 {user}  4096 Feb 10 08:21 .store
+-rwxr-xr-x 1 {user} 75632 Feb 10 08:21 botsay
+-rwxr-xr-x 1 {user} 75632 Feb 10 08:20 dotnetsay
+```
+
+## Drawbacks
+
+The ideal workflow for repositories accessing private NuGet feeds, such as Azure DevOps, involves installing the NuGet plugin as a .NET global tool. However, the current proposal suggests installing the plugin as a tool-path .NET tool. As mentioned in the technical explanation section, customers can opt to install NuGet plugins as a global tool instead of a tool-path tool. To do this, they need to set the `NUGET_DOTNET_TOOLS_PLUGIN_PATHS` environment variable. This variable should point to the location of the .NET Tool executable, which the NuGet Client tooling can invoke when needed.
+
+## Rationale and alternatives
+
+### NuGet commands to install credential providers
+
+[Andy Zivkovic](https://github.com/zivkan) kindly proposed an alternative design in [[Feature]: Make NuGet credential providers installable via the dotnet cli](https://github.com/NuGet/Home/issues/11325). The recommendation was developing a command like `dotnet nuget credential-provider install Microsoft.Azure.Artifacts.CredentialProvider`. Here are the advantages and disadvantages of this approach:
+
+**Advantages:**
+- As Andy suggested in the linked issue, we can utilize nuspec package types to enhance the discovery of credential providers. By introducing a new package type called `CredentialProvider`, the `dotnet nuget credential-provider list` command can filter by `packageType:CredentialProvider` on nuget.org. This will display a list of credential providers available on the feed, similar to the `dotnet list package` command. Although there are currently limited options for credential providers, customers would benefit from the ability to search by `packageType:CredentialProvider` on nuget.org.
+
+**Disadvantages:**
```

Comment:
Some risk of `dotnet tool` introducing breaking changes that negatively impact NuGet. Since the tool directory is intended to put on the PATH, it's a low risk, but if `dotnet tool` started writing non-executable files into the directory, that would impact how NuGet discovers and runs plugins at runtime.

Reply:
Added `There is some risk of the dotnet tool introducing breaking changes that could negatively impact NuGet. If the dotnet tool started writing non-executable files into the directory, it would affect how NuGet discovers and runs plugins at runtime. ` to the `Drawbacks` section of the current proposal.